dcp_prototype/backend/wrangling/migrations/scripts/migrate_dcp_one_to_dcp_two.py

```diff
@@ -91,51 +113,46 @@ def consume_file(prefix, bucket, filequeue, dataset_metadata):
         if filename is None:
             break
 
-        # sanity check that the file should be processes
-        if ".json" not in filename:
-            continue
-
         file_prefix = prefix + filename
 
         qsize = filequeue.qsize()
         if qsize % 1000 == 0 and qsize > 0:
             print(f"Working with JSON input file: {file_prefix}, queued size={qsize}")
         object = s3_client.get_object(Bucket=bucket, Key=file_prefix)
         object_body = object["Body"].read()
+        object_data = json.loads(object_body)
 
-        tsv_generated_data_frame = json_normalize(flatten(json.loads(object_body), separator="."))
-        entity_type = get_entity_type(filename)
-        for row in tsv_generated_data_frame.iterrows():
-            metadata_row = row[1]
-            dataset_metadata.parse_flattened_row_of_json(metadata_row, entity_type)
-
+        process_file_data(filename, object_data, dataset_metadata)
         filequeue.task_done()
 
 
-def generate_metadata_structure_from_s3_uri(s3_uri, num_threads):
+def generate_metadata_structure_from_s3_uri(s3_uri, num_threads, dataset_metadata):
     """
     Transforms a project's metadata into the DCP 2.0 metadata schema. Metadata
     exists primarily in JSON files in the given S3 bucket.
     """
 
-    dataset_metadata = OldDatasetMetadata(s3_uri=s3_uri)
-
-    BUCKET_NAME = urlparse(s3_uri).netloc
+    bucket_name = urlparse(s3_uri).netloc
     PREFIX = urlparse(s3_uri).path
     if PREFIX.startswith("/"):
         PREFIX = PREFIX[1:]
 
-    paginator = S3_CLIENT.get_paginator("list_objects")
-    page_iterator = paginator.paginate(Bucket=BUCKET_NAME, Prefix=PREFIX)
-    filtered_iterator = page_iterator.search("Contents[?contains(Key, `.json`)][]")
+    s3_client = boto3.client("s3")
+    paginator = s3_client.get_paginator("list_objects")
+    page_iterator = paginator.paginate(Bucket=bucket_name, Prefix=PREFIX)
 
     file_list = []
     print("Gather objects: ", end="", flush=True)
-    for object in filtered_iterator:
-        object_filename = object.get("Key")
-        file_list.append(object_filename.split("/")[-1])
-        if len(file_list) % 1000 == 0:
-            print(len(file_list), end=" ", flush=True)
+    stop = False
+    for page in page_iterator:
+        bucket_objects = page.get("Contents")
+        for object in bucket_objects:
+            object_filename = object.get("Key")
+            file_list.append(object_filename.split("/")[-1])
+            if len(file_list) % 1000 == 0:
+                print(len(file_list), end=" ", flush=True)
```

Comment:
Perhaps a more descriptive print statement?

Reply:
On the screen it will look this this (if there were 4444 files).

Gather objects: 1000 2000 3000 4000 4444

The individual print statement at 153 is just printing the number a space and flushing to the screen.